presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java

```diff
@@ -148,27 +143,18 @@ public HivePartitionResult getPartitions(SemiTransactionalHiveMetastore metastor
         List<String> partitionNames = getFilteredPartitionNames(metastore, tableName, partitionColumns, effectivePredicate);
 
         // do a final pass to filter based on fields that could not be used to filter the partitions
-        int partitionCount = 0;
-        ImmutableList.Builder<HivePartition> partitions = ImmutableList.builder();
-        for (String partitionName : partitionNames) {
-            Optional<Map<ColumnHandle, NullableValue>> values = parseValuesAndFilterPartition(partitionName, partitionColumns, partitionTypes, constraint);
-
-            if (values.isPresent()) {
-                if (partitionCount == maxPartitions) {
-                    throw new PrestoException(HIVE_EXCEEDED_PARTITION_LIMIT, format(
-                            "Query over table '%s' can potentially read more than %s partitions",
-                            hiveTableHandle.getSchemaTableName(),
-                            maxPartitions));
-                }
-                partitionCount++;
-                partitions.add(new HivePartition(tableName, partitionName, values.get(), buckets));
-            }
-        }
+
+        Iterable<HivePartition> partitionsIterable = () -> partitionNames.stream()
+                .map(partitionName -> parseValuesAndFilterPartition(partitionName, partitionColumns, partitionTypes, constraint))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(partitionNameAndValues -> new HivePartition(tableName, partitionNameAndValues.partitionName, partitionNameAndValues.values, buckets))
```

Comment:
Add getters (we avoid directly referencing fields even for a private class like this). Also, I'd use a shorter lambda argument name to improve readability.